src/runtime_src/core/edge/drm/zocl/zocl_ctrl_ert.c

```diff
@@ -870,22 +862,21 @@ static void zert_cmd_cfg_start(struct zocl_ctrl_ert *zert, struct xgq_cmd_sq_hdr
 static void zert_cmd_cfg_end(struct zocl_ctrl_ert *zert, struct xgq_cmd_sq_hdr *cmd,
 			     struct xgq_com_queue_entry *resp)
 {
-	int rc;
+	int rc = 0;
 
 	if (zert->zce_config_completed) {
 		zert_err(zert, "ERT is not in config mode");
 		init_resp(resp, cmd->cid, -EINVAL);
 		return;
 	}
-	zert->zce_config_completed = true;
 
 	zocl_get_zdev()->kds.cu_intr_cap = 1;
 	zocl_get_zdev()->kds.cu_intr = 1;
 	kds_cfg_update(&zocl_get_zdev()->kds);
 
+	zert->zce_config_completed = true;
+
 	rc = zert_validate_cus(zert);
```

Comment:
not sure why this is still needed... i thought this end of configure command should be a no-op now. especially, each cu is configured and unconfigured individually, why you still need to validate them as a group?

Reply:
Actually as per the existing code after end of the configuration we are updating the KDS. 
If we remove this part then after every CU/SCU configuration/un-configuration we need to update KDS. 
I am avoiding that many times KDS to update. Instead once the configuration is end then only I will update KDS.

Reply:
same here.